packages/suite/src/views/onboarding/steps/SelectBackupType.tsx

```diff
@@ -0,0 +1,346 @@
+import {
+    Badge,
+    Divider,
+    ElevationUp,
+    Icon,
+    Radio,
+    Text,
+    Warning,
+    useElevation,
+} from '@trezor/components';
+import {
+    Elevation,
+    borders,
+    mapElevationToBackground,
+    mapElevationToBorder,
+    spacings,
+    spacingsPx,
+    zIndices,
+} from '@trezor/theme';
+import { CSSProperties, ReactNode, forwardRef, useState } from 'react';
+import styled from 'styled-components';
+import { useSelector } from '../../../hooks/suite';
+import { selectDevice } from '@suite-common/wallet-core';
+import { DeviceModelInternal } from '@trezor/connect';
+import {
+    FloatingFocusManager,
+    autoUpdate,
+    useFloating,
+    useClick,
+    useDismiss,
+    useRole,
+    useInteractions,
+    FloatingPortal,
+    size,
+    offset,
+} from '@floating-ui/react';
+import { TranslationKey } from '@suite-common/intl-types';
+import { Translation } from '../../../components/suite';
+
+export const selectBackupTypes = [
+    'shamir-default',
+    'shamir-advance',
+    '12-words',
+    '24-words',
+] as const;
+
+export type BackupType = (typeof selectBackupTypes)[number];
+
+const isDeviceLegacyMap: Record<DeviceModelInternal, boolean> = {
+    [DeviceModelInternal.T1B1]: true,
+    [DeviceModelInternal.T2T1]: true,
+    [DeviceModelInternal.T2B1]: false,
+};
+
+const isLegacyDevice = (model?: DeviceModelInternal) =>
+    model !== undefined && isDeviceLegacyMap[model];
+
+export const isShamirBackupType = (type: BackupType) =>
+    ['shamir-default', 'shamir-advance'].includes(type);
+
+const typesToLabelMap: Record<BackupType, TranslationKey> = {
+    'shamir-default': 'TR_ONBOARDING_SEED_TYPE_SINGLE_SEED',
+    'shamir-advance': 'TR_ONBOARDING_SEED_TYPE_ADVANCED',
+    '12-words': 'TR_ONBOARDING_SEED_TYPE_12_WORDS',
+    '24-words': 'TR_ONBOARDING_SEED_TYPE_24_WORDS',
+};
+
+export const defaultBackupTypeMap: Record<DeviceModelInternal, BackupType> = {
```

Comment:
`DeviceModelInternal.T3T1` landed in the repo so now it's complaining that it doesn't have value for this property.

Reply:
:+1: